core/src/Cabana_AoSoA.hpp

```diff
@@ -358,8 +365,21 @@ class AoSoA
         // Assign the new capacity.
         _capacity = num_soa_alloc * vector_length;
 
-        // If we need more SoA objects then resize.
-        Kokkos::resize( _data, num_soa_alloc );
+        // We need more SoA objects so allocate a new view and copy the
+        // existing data.
+        soa_view resized_data(
+            Kokkos::ViewAllocateWithoutInitializing( _data.label() ),
+            num_soa_alloc );
+        if ( _num_soa > 0 )
+        {
+            // Note that we make an unmanaged view here. The subview approach
+            // was creating a view rank mismatch. This allows us to have two
+            // views of the same size for the call to deep_copy.
+            Kokkos::View<soa_type *, device_type, Kokkos::MemoryUnmanaged>
+                resized_data_um( resized_data.data(), _num_soa );
+            Kokkos::deep_copy( resized_data_um, _data );
+        }
```

Comment:
FYI https://github.com/kokkos/kokkos/pull/2285

Also
```C++
Kokkos::deep_copy(Kokkos::subview(resized_data, std::make_pair(0, _num_soa)), _data);
```

Reply:
that would have been nice - and yeah good call on the syntax